ztools/x509/json.go

```diff
@@ -228,37 +229,40 @@ func (c *Certificate) MarshalJSON() ([]byte, error) {
 	jc.SubjectKeyInfo.KeyAlgorithm = c.PublicKeyAlgorithm
 
 	// Include all subject names, DNS names there are
-	for _, obj := range c.Subject.Names {
-
-		switch name := obj.Value.(type) {
-		case string:
-
-			flag := false
-
-			if len(name) > 2 && name[0] == '*' {
-				flag = govalidator.IsURL(name[2:])
-			} else {
-				flag = govalidator.IsURL(name)
-			}
+	name := c.Subject.CommonName
+	isValid := false
+
+	// Check for wildcards and redacts, ignore malformed urls
+	if len(name) > 2 && name[0] == '*' && name[1] == '.' {
+		isValid = govalidator.IsURL(name[2:])
+	} else if len(name) > 2 && name[0] == '?' && name[1] == '.' {
+		isValid = isValid || govalidator.IsURL(name[2:])
+	} else {
+		isValid = govalidator.IsURL(name)
+	}
```

Comment:
Hmm, actually I'm realizing the logic isn't repeated exactly. For DNSNames we also check for TLDs